stdlib/public/runtime/Enum.cpp

```diff
@@ -154,12 +154,16 @@ swift::swift_getEnumCaseSinglePayload(const OpaqueValue *value,
     auto *valueAddr = reinterpret_cast<const uint8_t*>(value);
     auto *extraTagBitAddr = valueAddr + payloadSize;
     unsigned extraTagBits = 0;
-    // FIXME: endianness
     unsigned numBytes = getNumTagBytes(payloadSize,
                                        emptyCases-payloadNumExtraInhabitants,
                                        1 /*payload case*/);
 
+#if defined(__BIG_ENDIAN__)
+    small_memcpy(reinterpret_cast<uint8_t*>(&extraTagBits) + 4 - numBytes,
```

Comment:
It comes from the fact that the extra tag can only be 1 byte, 2 bytes or 4 bytes in size (according to the `getNumTagBytes` function). I agree that this limit can be stated more clearly in the code.